### DISCUSSION TITLE
Help with getServerSideSitemapIndex

### DISCUSSION THREAD

**Discussion by 4rt10m** [iamvishnusankar/next-sitemap#436] [Q&A]
Do we need to create both, "Server side index-sitemaps" and "server side sitemap"?

In the docs, the example of getServerSideSitemapIndex returns a static array of sitemap paths:
```
return getServerSideSitemapIndex(ctx, [
    'https://example.com/path-1.xml',
    'https://example.com/path-2.xml',
  ])
```

Do we need to create "server side sitemap" separately and add this sitemap url to getServerSideSitemapIndex? Or can we dynamically create all the sitemaps inside the getServerSideSitemapIndex? And how? 

I've read and reread the docs, but can't seem to get it. 

Thanks!

**Comment by kshitijb** [comment]
@4rt10m were you able to figure it out?

**Reply by 4rt10m** [reply]
I’ve created a sitemapindex.xml folder in pages with index.js file and used getServerSideSitemapIndex to get all my sitemaps.

And then created sitemaps folder, with folders for all my sitemaps like this: sitemapname.xml and index.js inside. E.g 
/pages/sitemaps/pages-sitemaps.xml/index.js

And created each individual sitemap like so:

```
import { getServerSideSitemap } from "next-sitemap";
import { getDirectusClient } from "../../../lib/directus";

export const getServerSideProps = async (ctx) => {
  // Method to source urls from cms
  const directus = await getDirectusClient();

  const pages = directus.items('pages');

  const urls = await pages.readByQuery({
    fields: ["slug", "parent", "date_updated"],
    filter: { status: { _eq: "published" } },
    limit: -1,
  });

  const fields = urls.data.map((item) => ({
    loc: `${process.env.NEXT_PUBLIC_URL}/${item.parent}/${item.slug}`,
    lastmod: item.date_updated,
    priority: 0.7,
    changefreq: "daily",
  }))

return getServerSideSitemap(ctx, fields);
}

// Default export to prevent next.js errors
export default function SitemapIndex() {}
```
